### PR TITLE
Wiring entity remove

### PR DESCRIPTION
**Pull request by RobotCaleb**
This is rebased on #265

Extracted wiring utility functions
wired up moving, removing, creating attaches on entities

### CODE REVIEW

**Comment by chrisforbes** [comment]
We will need similar removal support for attaches orphaned by surface or
attach removal, I think.

**Comment by chrisforbes** [comment]
That second patch does /all the things/ and is hard to follow. Can we split
it?

**Comment by RobotCaleb** [comment]
As requested.

**Comment by chrisforbes** [comment]
Surface or frame removal, I meant.
